packages/graphback-db-manage/tests/GraphQLSchemaManager.test.ts

```diff
@@ -0,0 +1,68 @@
+import { Change, ChangeType } from '@graphql-inspector/core';
+// tslint:disable-next-line: match-default-export-name no-implicit-dependencies
+import ava, { ExecutionContext } from 'ava';
+import { GraphQLSchemaManager, InMemoryModelProvider } from "../src/database";
+import { GraphQLSchemaChangeTypes } from '../src/database/migrations/GraphQLSchemaChangeTypes';
+
+const oldSchemaDefault = `
+type User {
+  id: ID!
+  name: String!
+}
+
+type Note {
+  id: ID!
+  title: String!
+}
+`;
+
+const currentSchemaDefault = `
+type User {
+  id: ID!
+  age: Int
+}
+
+type Test {
+  id: ID!
+  name: String
+}
+`;
+
+const setup = (oldSchemaText: string = oldSchemaDefault, newSchemaText: string = currentSchemaDefault) => {
+
+  const provider = new InMemoryModelProvider(oldSchemaText, newSchemaText);
+
+  const schemaManager = new GraphQLSchemaManager({ provider: provider });
+
+  return { provider, schemaManager };
+}
+
+
+ava('it should detect multiple changes', (t: ExecutionContext) => {
+  const { schemaManager } = setup();
+
+  const changes = schemaManager.getChanges();
+
+  t.assert(changes.length === 2);
+  // tslint:disable-next-line: no-null-keyword
+  t.snapshot(JSON.stringify(changes, null, 2));
+});
+
+ava('it should not have invalid change types', (t: ExecutionContext) => {
+  const { schemaManager } = setup();
+
+  const changes = schemaManager.getChanges();
+
+  const invalidChangeTypes = changes.filter((c: Change) => c === GraphQLSchemaChangeTypes[c.type]);
+
+  t.assert(invalidChangeTypes.length === 0);
+});
+
+ava('it should update the old schema to match the new one', (t: ExecutionContext) => {
+  const { schemaManager, provider } = setup();
+
+  schemaManager.updateOldSchema();
+
+  t.assert(provider.getCurrentSchemaText() === provider.getPreviousSchemaText());
+  t.snapshot(provider.getPreviousSchemaText());
```

Comment:
This snapshot makes no sense for me. Snapshots should be used to test outcomes. This is just snapshoting what we provided wich in turn going to be a maintenance nightmare and zero value.